src/components/DetailsPanel/InterpretationsUnit/RichTextEditor/RichTextEditor.js

```diff
@@ -13,53 +12,66 @@ import {
     colors,
 } from '@dhis2/ui'
 import PropTypes from 'prop-types'
-import React, { createRef, useState } from 'react'
-import classes from './styles/RichTextEditor.module.css'
-
-const smileyFace = ':-)'
-const sadFace = ':-('
-const thumbsUp = ':+1'
-const thumbsDown = ':-1'
+import React, { createRef, useEffect, useState } from 'react'
+import {
+    convertCtrlKey,
+    insertMarkdown,
+    emojis,
+    EMOJI_SMILEY_FACE,
+    EMOJI_SAD_FACE,
+    EMOJI_THUMBS_UP,
+    EMOJI_THUMBS_DOWN,
+} from './markdownHandler'
+import {
+    mainClasses,
+    toolbarClasses,
+    emojisPopoverClasses,
+} from './styles/RichTextEditor.style.js'
 
-const emoticonsButtonRef = createRef()
+const emojisButtonRef = createRef()
+const textareaRef = createRef()
```

Comment:
You should use `useRef` in the component function body. This creates a single ref for the module (~file) so this could get very problematic if there are multiple RickTextEditors in view at once.